File: SMarketPlace/SMarketPlace.Web/Modules/ServerTypes/JobMP/AsignaSucursalForm.ts
import { IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AsignaSucursalForm {
    IdMp: IntegerEditor;
    IdSucursal: IntegerEditor;
}

export class AsignaSucursalForm extends PrefixedContext {
    static formKey = 'JobMP.AsignaSucursal';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AsignaSucursalForm.init)  {
            AsignaSucursalForm.init = true;

            var w0 = IntegerEditor;

            initFormType(AsignaSucursalForm, [
                'IdMp', w0,
                'IdSucursal', w0
            ]);
        }
    }
}

File: SMarketPlace/SMarketPlace.Web/Modules/JobMP/AsignaSucursal/AsignaSucursalForm.cs
using Serenity.ComponentModel;

namespace SMarketPlace.JobMP.Forms;

[FormScript("JobMP.AsignaSucursal")]
[BasedOnRow(typeof(AsignaSucursalRow), CheckNames = true)]
public class AsignaSucursalForm
{
    public int IdMp { get; set; }
    public int IdSucursal { get; set; }
}